projects/core/src/cart/store/effects/cart.effect.ts

```diff
@@ -17,56 +11,127 @@ import { CartActions } from '../actions/index';
 
 @Injectable()
 export class CartEffects {
+  // TODO: remove when removing cart store module
+  @Effect()
+  loadCart2$: Observable<CartActions.LoadMultiCart> = this.actions$.pipe(
+    ofType(CartActions.LOAD_CART),
+    map(
+      (action: CartActions.LoadCart) =>
+        new CartActions.LoadMultiCart(action.payload)
+    )
+  );
+
+  // TODO: change to listen on LoadMultiCart action, remove old actions usage
   @Effect()
   loadCart$: Observable<
     | CartActions.LoadCartFail
+    | CartActions.LoadMultiCartFail
     | CartActions.LoadCartSuccess
+    | CartActions.LoadMultiCartSuccess
     | CartActions.ClearCart
+    | CartActions.ClearMultiCart
+    | CartActions.RemoveCart
   > = this.actions$.pipe(
     ofType(CartActions.LOAD_CART),
-    map(
-      (action: {
-        type: string;
-        payload?: { userId: string; cartId: string };
-      }) => action.payload
-    ),
+    map((action: CartActions.LoadCart) => action.payload),
     mergeMap(payload => {
+      // TODO: get rid of cartData
       const loadCartParams = {
         userId: (payload && payload.userId) || this.cartData.userId,
         cartId: (payload && payload.cartId) || this.cartData.cartId,
       };
 
       if (this.isMissingData(loadCartParams)) {
-        return of(new CartActions.LoadCartFail({}));
+        return from([
+          new CartActions.LoadCartFail({}),
+          new CartActions.LoadMultiCartFail({ cartId: loadCartParams.cartId }),
+        ]);
       }
       return this.cartConnector
         .load(loadCartParams.userId, loadCartParams.cartId)
         .pipe(
-          map((cart: Cart) => {
-            return new CartActions.LoadCartSuccess(cart);
+          mergeMap((cart: Cart) => {
+            if (loadCartParams.cartId === 'current' && cart) {
+              return [
+                new CartActions.LoadCartSuccess(cart),
+                new CartActions.LoadMultiCartSuccess({
+                  cart,
+                  userId: loadCartParams.userId,
+                  extraData: payload.extraData,
+                }),
+                new CartActions.RemoveCart('current'),
+              ];
+            } else if (cart) {
+              return [
+                new CartActions.LoadCartSuccess(cart),
+                new CartActions.LoadMultiCartSuccess({
+                  cart,
+                  userId: loadCartParams.userId,
+                  extraData: payload.extraData,
+                }),
+              ];
+            }
+            return [
```

Comment:
it's a case `if(!cart)`.
Does it ever happen that backend responds with an undefined cart without an HTTP error?

Reply:
For `current` cart call connector can return `null`.